op-node/sources/receipts.go

```diff
@@ -82,13 +91,387 @@ func makeReceiptRequest(txHash common.Hash) (*types.Receipt, rpc.BatchElem) {
 	}
 }
 
-// NewReceiptsFetcher creates a receipt fetcher that can iteratively fetch the receipts matching the given txs.
-func NewReceiptsFetcher(block eth.BlockID, receiptHash common.Hash, txHashes []common.Hash, getBatch BatchCallContextFn, batchSize int) eth.ReceiptsFetcher {
-	return NewIterativeBatchCall[common.Hash, *types.Receipt, types.Receipts](
-		txHashes,
-		makeReceiptRequest,
-		makeReceiptsFn(block, receiptHash),
-		getBatch,
-		batchSize,
-	)
+// Cost break-down sources:
```

Comment:
A lot of code here. I think we want mocks of each endpoint

Reply:
I think we should test this by having a set of optional tests that take environment variables for actual RPC endpoints from each provider. That way we can make real queries, and if the provider behavior changes we'll know quickly.

Reply:
Implemented the mocks / testing in #4561